sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/pages/ReportsEditPage.java

```diff
@@ -1127,19 +1127,18 @@ public IModel getModel(Object value) {
 			Locator.getFacade().getSiteService().getSiteTypes().stream().map(s -> "!site.template." + s).forEach(siteIdWithRoles::add);
 		}
 
-		List<String> roles = new ArrayList<String>();
-		try {
-			for (String s : siteIdWithRoles) {
+		Set<String> roles = new HashSet<String>();
+		for (String s : siteIdWithRoles) {
+			try {
 				Set<Role> roleSet = Locator.getFacade().getAuthzGroupService().getAuthzGroup(s).getRoles();
 				for (Role r : roleSet) {
 					roles.add(r.getId());
 				}
+			} catch (GroupNotDefinedException e) {
+				log.warn("AuthzGroup does not exist, skipping: {}", s);
```

Comment:
Are we sure you want to see these as warnings in the log? how about debug?

Reply:
agree